lib/engine/ability/train_discount.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class TrainDiscount < Base
+      attr_reader :discount, :trains
+      def setup(discount:, trains:)
+        @discount = discount
+        @trains = trains
+      end
+
+      def discounted_price(train, price)
+        return price unless @trains.include?(train.name)
+
+        price - (discount > 1 ? discount : (price * discount))
```

Comment:
might be safer to do discount.is_a?(Integer), not sure test it

Reply:
Done

Reply:
But it did not work...  If I check which type/class discount has Ruby says Number.

Anyway, I changed back to using the > operator.